### PR TITLE
create list_dataset api and tests

### PR DESCRIPTION
**Pull request by cecilia-uu**
### What problem does this PR solve?

This PR have completed both HTTP API and Python SDK for 'list_dataset". In addition, there are tests for it.

### Type of change

- [x] New Feature (non-breaking change which adds functionality)

### CODE REVIEW

File: api/apps/dataset_api.py [hunk 1]
```diff
@@ -115,6 +115,22 @@ def create_dataset():
     except Exception as e:
         return construct_error_response(e)
 
+# -----------------------------list a datasets-------------------------------------------------------
+@manager.route('/', methods=['GET'])
+@login_required
+def list_datasets():
+    page_number = request.args.get("page", 1)
```
Comment (JinHai-CN) [inline api/apps/dataset_api.py]: 'page' and 'page_size' can be updated to 'dataset_offset' and 'dataset_count'

Comment (JinHai-CN) [inline api/apps/dataset_api.py]: 'page' and 'page_size' can be updated to 'dataset_offset' and 'dataset_count'

Comment (JinHai-CN) [inline api/apps/dataset_api.py]: We should allow user to get all of the datasets.

File: sdk/__init__.py
Comment (JinHai-CN) [inline sdk/__init__.py:1]: Remove it.

File: sdk/python/__init__.py
Comment (JinHai-CN) [inline sdk/python/__init__.py:1]: Remove it.

Comment (JinHai-CN) [inline sdk/python/__init__.py:1]: Remove it.

File: sdk/python/ragflow/__init__.py [hunk 2]
```diff
@@ -1,3 +1,5 @@
 import importlib.metadata
 
 __version__ = importlib.metadata.version("ragflow")
+
+from .ragflow import RAGFlow
```
Comment (JinHai-CN) [inline sdk/python/ragflow/__init__.py:5]: ???

File: sdk/python/ragflow/ragflow.py [hunk 3]
```diff
@@ -36,16 +39,48 @@ def create_dataset(self, dataset_name):
...
 
-    def list_dataset(self):
-        response = requests.get(self.dataset_url)
-        print(response)
-        if response.status_code == 200:
-            return response.json()['datasets']
-        else:
-            return None
+    # def list_dataset(self):
+    #     print("here ------------------------")
```
Comment (JinHai-CN) [inline sdk/python/ragflow/ragflow.py]: Remove comments

File: sdk/python/test/test_dataset.py [hunk 4]
```diff
@@ -15,12 +16,42 @@ def test_create_dataset(self):
...
+        # Call the list_datasets method
+        response = ragflow.list_dataset()
+        print(response)
+
+        # Check if the response is a dictionary and has the 'message' key
+        if isinstance(response, dict) and 'message' in response:
+            # Access the message using the 'message' key
+            assert response['message'] == "attempt to list datasets"
+        else:
+            print("Unexpected response format:", response)
```
Comment (JinHai-CN) [inline sdk/python/test/test_dataset.py]: Fail branch need raise exception

File: sdk/python/test/test_dataset.py [hunk 5]
```diff
@@ -15,12 +16,42 @@ def test_create_dataset(self):
...
+
+        # If the response is a dictionary with a 'data' key, proceed with the test
+        if 'data' in response:
+            datasets = response['data']
+            assert len(datasets) > 0
+            print(datasets)
```
Comment (JinHai-CN) [inline sdk/python/test/test_dataset.py]: Check list size and each dataset name is correct.

File: sdk/python/ragflow/__init__.py (hunk 2 above)
Comment (cecilia-uu) [reply sdk/python/ragflow/__init__.py:5]: It means that I import this class from the same level of directory.

File: sdk/python/test/test_dataset.py [hunk 6]
```diff
@@ -15,12 +16,53 @@ def test_create_dataset(self):
         4. update the kb
         5. delete the kb
         '''
-        ragflow = RAGFLow(API_KEY, HOST_ADDRESS)
+
 
         # create a kb
         res = ragflow.create_dataset("kb1")
+        print(res)
```
Comment (JinHai-CN) [inline sdk/python/test/test_dataset.py]: Remove the print

File: sdk/python/test/test_dataset.py [hunk 7]
```diff
@@ -15,12 +16,53 @@ def test_create_dataset(self):
...
+        dataset_name = res['data']['dataset_name']
+        print(dataset_name)
+
+    def test_list_dataset_success(self):
+        # Call the list_datasets method
+        response = ragflow.list_dataset()
+
+        code, datasets = response
+        assert len(datasets) > 0
+
```
Comment (JinHai-CN) [inline sdk/python/test/test_dataset.py:31]: This isn't a qualified test case

**Review by JinHai-CN — CHANGES_REQUESTED** [review]
Test cases are not enough.

**Comment by JinHai-CN** [comment]
Please add these test cases in dataset_test.py:
1. No any knowledge base, expect list API to get empty result.
2. Create 100 knowledge base, expect list API will return 100 result with expected name.

**Pull request merged**